R/parse-body.R

```diff
@@ -191,12 +192,14 @@ parser_multi <- function() {
     boundary <- stri_match_first_regex(content_type, "boundary=([^; ]{2,})", case_insensitive = TRUE)[,2]
     toparse <- parse_multipart(value, boundary)
     # content-type detection
-    lapply(toparse, function(x) {
+    args <- lapply(toparse, function(x) {
       if (!is.null(x$filename)) {
         x$content_type <- getContentType(tools::file_ext(x$filename))
       }
       parseRaw(x)
     })
+    # combine together args that share the same name
+    combine_keys(args, FALSE)
```

Comment:
Same as the json one above. Seems odd that a multi part would combine things and return a different shape only if the names are duplicated. Nothing wrong having a list with two items of the same name.

Reply:
Or is this where the magic is happening to be able to return `list(file1 = val1, file2 = val2, ....)`?

Reply:
> 
> 
> Or is this where the magic is happening to be able to return `list(file1 = val1, file2 = val2, ....)`?

Yes, for an arbitrary number of files. I think we discussed that all files should be returned in the form list(filename = raw).

It ends up in the args list as args = list(arg1 = 1, argfile1 = list(filename1 = raw1, ...))

Reply:
> If the value of a form field is a set of files rather than a single file, that value MUST be transmitted by supplying each in a separate part, but all with the same name parameter. 

https://tools.ietf.org/id/draft-ietf-appsawg-multipart-form-data-01.html

This is specific to the multipart content-type. I removed it from the json parser.